kernel/src/sync/event_flags.rs

```diff
@@ -0,0 +1,514 @@
+// Copyright (c) 2025 vivo Mobile Communication Co., Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::SpinLock;
+use crate::{
+    error::{code, Error},
+    irq, scheduler,
+    scheduler::{InsertMode, WaitQueue},
+    thread,
+    thread::Thread,
+    time::WAITING_FOREVER,
+    types::ArcList,
+};
+use bitflags::bitflags;
+use core::cell::Cell;
+type ThreadList = ArcList<Thread, thread::OffsetOfSchedNode>;
+
+bitflags! {
+    #[derive(Default, Debug, Clone, Copy)]
+    pub struct EventFlagsMode: u8 {
+        const ANY = 1 << 0;
+        const ALL = 1 << 1;
+        const NO_CLEAR = 1 << 2;
+    }
+}
+
+#[derive(Debug)]
+pub struct EventFlags {
+    // We let the Spinlock protect the whole semaphore.
+    pending: SpinLock<WaitQueue>,
+    flags: Cell<u32>,
+}
+
+impl Default for EventFlags {
+    fn default() -> Self {
+        Self::const_new()
+    }
+}
+
+impl EventFlags {
+    pub const fn const_new() -> Self {
```

Comment:
```suggestion
    pub const fn new() -> Self {
```
We use `const_new` when we have two versions of constructor, `const_new` for static construction, `new` for runtime construction. If `new` is already `const`, there is no need to add `const_new`.